### PR TITLE
DAOS-16365 client: use D_ASPRINTF and fix format issues

### PR DESCRIPTION
**Pull request by wiliamhuang**
Fix a few minor remaining issues in previous PR intercepting dlsym and zeInit.
1) Use D_ASPRINTF when possible
2) Remove unneeded newline, debugging output, and parentheses.

Features: pil4dfs

Required-githooks: true
Skipped-githooks: codespell

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by wiliamhuang** [comment]
This PR is a follow-up of https://github.com/daos-stack/daos/pull/14932 to fix a few minor issues.

File: src/client/dfuse/pil4dfs/hook.c [hunk 1]
```diff
@@ -306,28 +306,17 @@ determine_lib_path(void)
...
-	if (rc >= PATH_MAX) {
-		free(path_libdl);
-		path_libdl = NULL;
-		DS_ERROR(ENAMETOOLONG, "path_libdl is too long");
-		goto err_1;
+		goto err;
 	}
+	D_ASPRINTF(path_libdl, "%s/libdl-%s.so", lib_dir_str, libc_version_str);
+	if (path_libdl == NULL)
+		goto err;
```
Comment (tanabarr) [inline src/client/dfuse/pil4dfs/hook.c:319]: don't you need another strnlen PATH_MAX check here?

**Review by tanabarr — APPROVED** [review]

File: src/client/dfuse/pil4dfs/hook.c (hunk 1 above)
Comment (wiliamhuang) [reply src/client/dfuse/pil4dfs/hook.c:319]: This path is shorter than the above one, "%s/libpthread-%s.so", whose length was checked already.

**Comment by wiliamhuang** [comment]
@knard-intel @mchaarawi Could you please review? I have another PR waiting for this one. Thank you very much!

**Review by knard-intel — APPROVED** [review]

**Review by mchaarawi — CHANGES_REQUESTED** [review]
no requesting changes, seeking clarification before a gatekeeper lands

File: src/client/dfuse/pil4dfs/hook.c [hunk 2]
```diff
@@ -306,28 +306,17 @@ determine_lib_path(void)
...
-		DS_ERROR(ENOMEM, "Failed to allocate memory for path_libpthread");
-		goto err_1;
-	}
-	if (rc >= PATH_MAX) {
-		free(path_libpthread);
-		path_libpthread = NULL;
+	D_ASPRINTF(path_libpthread, "%s/libpthread-%s.so", lib_dir_str, libc_version_str);
+	if (path_libpthread == NULL)
+		goto err;
+	if (strnlen(path_libpthread, PATH_MAX) >= PATH_MAX) {
```
[the remark this answers is not in the document]
Comment (wiliamhuang) [reply src/client/dfuse/pil4dfs/hook.c:312]: I agree. "==" would be more accurate.

File: src/client/dfuse/pil4dfs/hook.c [hunk 3]
```diff
@@ -306,28 +306,17 @@ determine_lib_path(void)
...
-		goto err_1;
-	}
-	if (rc >= PATH_MAX) {
-		free(path_libdl);
-		path_libdl = NULL;
-		DS_ERROR(ENAMETOOLONG, "path_libdl is too long");
-		goto err_1;
+		goto err;
 	}
+	D_ASPRINTF(path_libdl, "%s/libdl-%s.so", lib_dir_str, libc_version_str);
```
[the remark this answers is not in the document]
Comment (wiliamhuang) [reply src/client/dfuse/pil4dfs/hook.c:317]: Yes. We checked the string length above for "%s/libpthread-%s.so" already.

[the remark this answers is not in the document]
Comment (mchaarawi) [reply src/client/dfuse/pil4dfs/hook.c:317]: but this is a different string?

File: src/client/dfuse/pil4dfs/int_dfs.c [hunk 4]
```diff
@@ -1198,7 +1197,6 @@ new_dlsym_c(void *handle, const char *symbol)
 {
 	if (!d_hook_enabled)
 		goto org_dlsym;
-	printf("Inside my dlsym().\n");
```
[the remark this answers is not in the document]
Comment (wiliamhuang) [reply src/client/dfuse/pil4dfs/int_dfs.c:1201]: Right. This is in 2.6. This code is only compiled on non-x86_64 CPU.

File: src/client/dfuse/pil4dfs/hook.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (mchaarawi) [reply src/client/dfuse/pil4dfs/hook.c:317]: nevermind. i see the prefix is the same, and you attach something shorter

File: src/client/dfuse/pil4dfs/int_dfs.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (mchaarawi) [reply src/client/dfuse/pil4dfs/int_dfs.c:1201]: ok. we should add a commit hook probably to catch such issues before they land in the first place.

**Review by mchaarawi — APPROVED** [review]

**Pull request merged**